R/step-subset-separate.R

```diff
@@ -50,7 +50,7 @@ separate.dtplyr_step <- function(data, col, into,
     abort("`sep` must be a character vector.")
   }
 
-  col <- enexpr(col)
+  col <- quo_squash(enexpr(col))
```

Comment:
To allow for numeric arguments, and only those coercible to integer (e.g. not `1.5`), we could add after this

```R
if (is.numeric(col)) {                                                           
  col <- as.symbol(vctrs::vec_slice(data$vars, col))                             
} 
```